src/main/java/bookface/model/book/Book.java

```diff
@@ -18,15 +18,33 @@ public class Book {
     private boolean isLoaned = false;
     private Person loanee = null;
 
+    private String loanStatus;
```

Comment:
To represent this idea of being possibly on loan, we can use subclasses (`AvailableBook` and `BookOnLoan`) and the Factory Design Pattern. So this avoids the possibility of having a NullPointerException, the need for additional `isLoaned` and `loanStatus` fields, and if-else checks. Then if we want to add more subclasses to track state, such as `OverdueBook`, we can do so easily.

Reply:
An issue is that we have to replace the object in the List, but I think that's doable

Reply:
I was trying to avoid making subclasses and making it overly complicated but you're right. I will try to refactor everything one more time. As a compromise can we make it a single subclass (eg. BookAvailability) instead of two?

Reply:
Oh yea good point, let's push it back to v1.3